types/src/primary.rs

```diff
@@ -527,21 +527,36 @@ pub struct BatchMessage {
     pub transactions: Batch,
 }
 
+pub type BlockRemoverResult<T> = Result<T, BlockRemoverError>;
+
+#[derive(Clone, Debug)]
+pub struct BlockRemoverError {
+    pub ids: Vec<CertificateDigest>,
+    pub error: BlockRemoverErrorType,
+}
+
+// TODO: refactor BlockError & BlockRemoverError to be one type shared by get/remove collections.
+#[derive(Clone, Debug, PartialEq)]
+pub enum BlockRemoverErrorType {
```

Comment:
Nit: Traditionally in Rust codebases, this would be named `BlockRemoverErrorKind`.